### ISSUE TITLE
Assertion failed 'structType != NO_CLASS_HANDLE' in 'System.SpanTests.IndexerTests:IndexerWithRangeTest()

### ISSUE THREAD

**Issue by echesakovMSFT** [dotnet/coreclr#20958]
https://ci.dot.net/job/dotnet_coreclr/job/master/job/jitstress/job/x64_checked_ubuntu_corefx_baseline/751/
https://ci.dot.net/job/dotnet_coreclr/job/master/job/jitstress/job/arm64_cross_checked_ubuntu16.04_corefx_baseline_tst/37/
https://ci.dot.net/job/dotnet_coreclr/job/master/job/jitstress/job/arm_cross_checked_ubuntu_corefx_baseline_tst/28/

```
Assert failure(PID 2702 [0x00000a8e], Thread: 3226 [0x0c9a]): Assertion failed 'structType != NO_CLASS_HANDLE' in 'System.SpanTests.IndexerTests:IndexerWithRangeTest()' (IL size 204)

    File: /mnt/j/workspace/dotnet_coreclr/master/jitstress/arm_cross_checked_ubuntu_corefx_baseline/src/jit/importer.cpp Line: 768
    Image: /ssd/j/workspace/dotnet_coreclr/master/jitstress/arm_cross_checked_ubuntu_corefx_baseline_tst/_/fx/artifacts/bin/testhost/netcoreapp-Linux-Release-arm/dotnet
```

/cc @dotnet/jit-contrib 

The test assembly was excluded from Linux/arm and Linux/arm64 jobs in https://github.com/dotnet/coreclr/pull/20962
When this is fixed please update **tests/arm/corefx_linux_test_exclusions.txt** and **tests/arm64/corefx_linux_test_exclusions.txt** files

**Comment by CarolEidt** [comment]
If no one else is looking at this already I can take a look at it.

**Comment by CarolEidt** [comment]
@echesakovMSFT - I can't reproduce this on Linux/x64, which was the first failure above. However, I notice that you only added exclusions for arm and arm64 - am I missing something? Did the failure simply go away on Linux/x64?

**Comment by echesakovMSFT** [comment]
I see that the assertion failure happened during last corefx baseline [Build #754 (Nov 14, 2018 4:26:06 AM)](https://ci.dot.net/job/dotnet_coreclr/job/master/job/jitstress/job/x64_checked_ubuntu_corefx_baseline/754/) 

The reason why I disabled only arm and arm64 is that we don't have similar exclusion mechanism for x64 corefx outerloop jobs.

The failing test was added recently https://github.com/dotnet/corefx/pull/33352 so if you corefx tests build is older than 4 days you need to rebuild it.

**Comment by CarolEidt** [comment]
Oops, operator error. I failed to set `COMPlus_TieredCompilation=0`. I am now able to reproduce it. Thanks!

**Comment by AndyAyersMS** [comment]
@CarolEidt we can close this now, right?

**Comment by CarolEidt** [comment]
Yes, sorry. It probably should have been closed with the actual fix from @jkotas #21048
